src/components/TasksList/TasksList.tsx

```diff
@@ -27,13 +93,53 @@ const TasksList: React.FC = () => {
           <Tab label='My tasks' />
         </Tabs>
       </AppBar>
-      <TabPanel value={value} index={0}>
-        Newbie Tasks Component
+
+      {updateTaskLoading && <LinearProgress />}
+
+      <TabPanel p={2} value={tabIndex} index={0}>
+        {loading && <PlaceHolderTaskList />}
+        {!loading && data && data.newbie.newbieTasks.length === 0 && (
```

Comment:
This is pretty long condition. I think it could be written in more elegant way being more readable and follows DRY.
I thinking about some kind of extra abstraction here:

```
  const getTabPanelDataByRole = (data: any, role: UserRole) => {
    const isEmpty = (tasks: []) => tasks.length === 0;
    const { newbieTasks } = data && data.newbie;
    if (isNewbie(role)) {
      return {
        tasks: newbieTasks,
        isEmpty: isEmpty(newbieTasks),
      };
    } else if (isBuddy(role)) {
      const { buddyTasks } = data && data.newbie;
      return {
        tasks: buddyTasks,
        isEmpty: isEmpty(buddyTasks),
      };
    } else {
      return {
        tasks: [],
        isEmpty: true,
      };
    }
  };

  const NewbieTab = getTabPanelDataByRole(data, UserRole.Newbie);
  const BuddyTab = getTabPanelDataByRole(data, UserRole.Buddy);
```

 that could be used inside JSX:

```
      <TabPanel p={2} value={tabIndex} index={0}>
        {loading && <PlaceHolderTaskList />}
        {!loading && NewbieTab.isEmpty && <EmptyTaskList />}
        {!loading && !NewbieTab.isEmpty && (
          <TaskTabsContent onChange={onTaskChange} tasks={NewbieTab.tasks} />
        )}
      </TabPanel>
      <TabPanel p={2} value={tabIndex} index={1}>
        {loading && <PlaceHolderTaskList />}
        {!loading && BuddyTab.isEmpty && <EmptyTaskList />}
        {!loading && !BuddyTab.isEmpty && (
          <TaskTabsContent onChange={onTaskChange} tasks={BuddyTab.tasks} />
        )}
      </TabPanel>
```

_This is only an quick example, you can think about sth better_ 😀

Reply:
fixed here https://github.com/BuildIt-Poland/buddy-app/pull/40/commits/4f2356f853516fb7efc33049cff09e1a8025e585